src/Grpc.Net.Client.Web/Internal/GrpcWebResponseStream.cs

```diff
@@ -56,6 +56,17 @@ public override async ValueTask<int> ReadAsync(Memory<byte> data, CancellationTo
         switch (_state)
         {
             case ResponseState.Ready:
+                if (data.Length == 0)
+                {
+                    // Handle zero byte reads.
+                    return 0;
```

Comment:
Shouldn't we try a zero-length read on `_inner` before immediately returning 0? It looks like this logic has the risk stalling in `TryReadDataAsync` during the next call to `ReadAsync` which will result in a 4 KB buffer or larger being rented even though it will be sliced to only 5 bytes for the header.